pkg/router/template/template_helper.go

```diff
@@ -321,41 +317,38 @@ func generateHAProxyMap(name string, td templateData) []string {
 
 // clipHAProxyTimeoutValue prevents the HAProxy config file
 // from using timeout values specified via the haproxy.router.openshift.io/timeout
-// annotation that exceed the maximum value allowed by HAProxy.
-// Return the parameter string instead of an err in the event that a
+// annotation that exceed the maximum value allowed by HAProxy, or by time.ParseDuration.
+// Return the empty string instead of an err in the event that a
 // timeout string value is not parsable as a valid time duration.
 func clipHAProxyTimeoutValue(val string) string {
 	// If the empty string is passed in,
 	// simply return the empty string.
 	if len(val) == 0 {
 		return val
 	}
-	endIndex := len(val) - 1
-	maxTimeout, err := time.ParseDuration(haproxyMaxTimeout)
+
+	// First check to see if the timeout will fit into a time.Duration
+	duration, err := templateutil.ParseHAProxyDuration(val)
 	if err != nil {
-		return val
-	}
-	// time.ParseDuration doesn't work with days
-	// despite HAProxy accepting timeouts that specify day units
-	if val[endIndex] == 'd' {
-		days, err := strconv.Atoi(val[:endIndex])
-		if err != nil {
-			return val
-		}
-		if maxTimeout.Hours() < float64(days*24) {
+		switch err.(type) {
+		case templateutil.OverflowError:
 			log.V(7).Info("Route annotation timeout exceeds maximum allowable by HAProxy, clipping to max")
-			return haproxyMaxTimeout
-		}
-	} else {
-		duration, err := time.ParseDuration(val)
-		if err != nil {
-			return val
-		}
-		if maxTimeout.Milliseconds() < duration.Milliseconds() {
-			log.V(7).Info("Route annotation timeout exceeds maximum allowable by HAProxy, clipping to max")
-			return haproxyMaxTimeout
+			return templateutil.HaproxyMaxTimeout
+		case templateutil.InvalidInputError:
+			log.V(7).Info("Route annotation timeout removed because input is invalid: " + err.Error())
```

Comment:
Use structured logging; 

```suggestion
			log.V(7).Info("Route annotation timeout removed because input is invalid", "error", err.Error())
```

Reply:
For that matter, we can use `log.Error` here:
```suggestion
			log.Error(err, "route annotation timeout removed because input is invalid")
```

Reply:
fixed for all of these